python/thirdweb-ai/src/thirdweb_ai/services/insight.py

```diff
@@ -5,25 +5,54 @@
 
 
 class Insight(Service):
-    def __init__(self, secret_key: str, chain_id: int | list[int]):
-        super().__init__(base_url="https://insight.thirdweb.com/v1", secret_key=secret_key)
+    def __init__(self, secret_key: str, chain_id: int | str | list[int | str]):
+        super().__init__(
+            base_url="https://insight.thirdweb.com/v1", secret_key=secret_key
+        )
         self.chain_ids = chain_id if isinstance(chain_id, list) else [chain_id]
+        self.chain_ids = self._normalize_chain_ids(self.chain_ids)
+
+    def _normalize_chain_ids(
```

Comment:
let's move this to a util file and apply the same logic for other services like Engine as well